x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvMedianAbsoluteDeviation.java

```diff
@@ -0,0 +1,357 @@
+        longs.count = 0;
+        return mad;
+    }
+
+    /**
+     * If the values are ascending pick the middle value or average the two middle values together.
+     */
+    static int ascending(Longs longs, IntBlock values, int firstValue, int count) {
+        if (longs.values.length < count) {
+            longs.values = ArrayUtil.grow(longs.values, count);
+        }
+        int middle = firstValue + count / 2;
+        long median = count % 2 == 1 ? values.getInt(middle) : avgWithoutOverflow(values.getInt(middle - 1), values.getInt(middle));
+        for (int i = 0; i < count; i++) {
+            long value = values.getInt(firstValue + i);
+            longs.values[i] = value > median ? Math.subtractExact(value, median) : Math.subtractExact(median, value);
+        }
+        return Math.toIntExact(longMedianOf(longs.values, count));
+    }
+
+    static int single(int value) {
+        return 0;
+    }
+
+    @MvEvaluator(extraName = "Long", finish = "finish", ascending = "ascending", single = "single")
+    static void process(Longs longs, long v) {
+        if (longs.values.length < longs.count + 1) {
+            longs.values = ArrayUtil.grow(longs.values, longs.count + 1);
+        }
+        longs.values[longs.count++] = v;
+    }
+
+    static long finish(Longs longs) {
+        long median = longMedianOf(longs.values, longs.count);
+        for (int i = 0; i < longs.count; i++) {
+            long value = longs.values[i];
+            // From here, this array contains unsigned longs
+            longs.values[i] = unsignedDifference(value, median);
+        }
+        long mad = NumericUtils.unsignedLongAsLongExact(unsignedLongMedianOf(longs.values, longs.count));
+        longs.count = 0;
+        return mad;
+    }
+
+    /**
+     * If the values are ascending pick the middle value or average the two middle values together.
+     */
+    static long ascending(Longs longs, LongBlock values, int firstValue, int count) {
+        if (longs.values.length < count) {
+            longs.values = ArrayUtil.grow(longs.values, count);
+        }
+        int middle = firstValue + count / 2;
+        long median = count % 2 == 1 ? values.getLong(middle) : avgWithoutOverflow(values.getLong(middle - 1), values.getLong(middle));
+        for (int i = 0; i < count; i++) {
+            long value = values.getLong(firstValue + i);
+            // From here, this array contains unsigned longs
+            longs.values[i] = unsignedDifference(value, median);
+        }
+        return NumericUtils.unsignedLongAsLongExact(unsignedLongMedianOf(longs.values, count));
+    }
+
+    static long single(long value) {
+        return 0L;
+    }
+
+    static long longMedianOf(long[] values, int count) {
+        // TODO quickselect
+        Arrays.sort(values, 0, count);
+        int middle = count / 2;
+        return count % 2 == 1 ? values[middle] : avgWithoutOverflow(values[middle - 1], values[middle]);
+    }
```

Comment:
In ascending cases, the Longs/Doubles object doesn't have data, and we artificially fill it without increasing the count (As we would have to reset it at the end again).

Anyway, after writing this, I started to think that it's indeed simpler to just use the full object instead of being "juggling" with values... So just changed it for simplicity